### PR TITLE
chore(extraction-plugin): redo chunk attachment

### PR DESCRIPTION
**Pull request by layershifter**
This PR redoes chunk management in the plugin to prevent "phantom chunks" that we were creating before.

- Griffel chunk will be created only if needed
- A hack with `GriffelDummyModule` was removed
- We will reuse `SplitChunksPlugin` if it's enabled to avoid redundant module movements
- New fixture added for `optiminizatin.splitChunks = false`

### CODE REVIEW

File: packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.ts [hunk 1]
```diff
@@ -115,48 +107,50 @@ export class GriffelCSSExtractionPlugin {
...
+        // Performs module movements if SplitChunksPlugin is not present
+        compilation.hooks.optimizeChunks.tap({ name: PLUGIN_NAME, stage: OPTIMIZE_CHUNKS_STAGE_ADVANCED }, () => {
+          moveCSSModulesToGriffelChunk(compilation);
+        });
+      }
 
+      // In this case we detach Griffel chunk from all other chunks and connect it with the main entrypoint
       if (this.attachToMainEntryPoint) {
-        compilation.hooks.afterChunks.tap(PLUGIN_NAME, () => {
+        compilation.hooks.optimizeChunks.tap({ name: PLUGIN_NAME, stage: 10 }, () => {
```
Comment (ling1726) [inline packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.ts:130]:
```suggestion
        compilation.hooks.optimizeChunks.tap({ name: PLUGIN_NAME, stage:  OPTIMIZE_CHUNKS_STAGE_ADVANCED}, () => {
```

**Review by ling1726 — APPROVED** [review]

File: packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.ts [hunk 2]
```diff
@@ -115,48 +107,50 @@ export class GriffelCSSExtractionPlugin {
...
-        compilation.chunkGraph.disconnectChunkAndModule(griffelChunk, dummyModule);
-      });
-
       compilation.hooks.processAssets.tap(
         {
           name: PLUGIN_NAME,
           stage: Compilation.PROCESS_ASSETS_STAGE_PRE_PROCESS,
         },
         assets => {
+          const griffelChunk = compilation.namedChunks.get('griffel');
```
Comment (ling1726) [inline packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.ts:148]: is the check for CSS assets under this new early return still necessary? there should always be assets if a griffel chunk exists?

Comment (layershifter) [reply packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.ts:148]: I would keep it to make TS happy and `.find()` can return `undefined`

File: packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.ts [hunk 3]
```diff
@@ -115,48 +107,50 @@ export class GriffelCSSExtractionPlugin {
...
+      };
+    }
+
     compiler.hooks.compilation.tap(PLUGIN_NAME, compilation => {
-      // A chunk where we will move all CSS modules
-      const griffelChunk = compilation.addChunk('griffel');
-      // Set up a dummy module to prevent the chunk from getting cleaned up by RemoveEmptyChunksPlugin
-      const dummyModule = new GriffelDummyModule();
+      if (!compiler.options.optimization.splitChunks) {
+        // Performs module movements if SplitChunksPlugin is not present
```
[the remark this answers is not in the document]
Comment (layershifter) [reply packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.ts]: I update all comments in this file, should be clearer now.

File: packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.ts (hunk 1 above)
Comment (layershifter) [reply packages/webpack-extraction-plugin/src/GriffelCSSExtractionPlugin.ts]: I moved these taps into a single call.

**Pull request merged**